ocaml/typing/ctype.ml

```diff
@@ -5838,7 +5869,8 @@ let rec normalize_type_rec visited ty =
         begin match !nm with
         | None -> ()
         | Some (n, v :: l) ->
-            if deep_occur ty (newgenty (Ttuple l)) then
+            let fake_ttuple = newgenty (Ttuple (List.map (fun ty -> None, ty) l)) in
+            if deep_occur ty fake_ttuple then
```

Comment:
Hmm - I guess we should rebase this whole branch onto the latest flamba-backend to get the `deep_occur_list` changes.  I can help with that later this week.